### ISSUE TITLE
We need our own IMessageBox implementation

### ISSUE THREAD

**Issue by retailcoder** [rubberduck-vba/Rubberduck#2941]
We're currently injecting a thin wrapper around `Microsoft.Windows.Forms.MessageBox` wherever we need an `IMessageBox`.

That gets the job done, but let's make our own message dialog (with the little ducky icon!) - that way we can have this:

 - [x] Thanks, please don't show me this message again

And store `DontShowThisAgain` state in the user's settings (no need for a settings page in the settings dialog though).

**Comment by Hosch250** [comment]
Uh, duckling? This is a level 2 issue at least, possibly even level 3.

First, UI. Big deal.
Second, modify the settings. A little more tricky--more UI and updating a bunch of settings files to ensure it is serialized/deserialized properly.
Third, figuring out which message box was checked and which one wasn't. Perhaps this "ID" can be based on the calling class + which call it is in the class (some classes call multiple times).

**Comment by retailcoder** [comment]
Why not just use the message's resource key?

**Comment by Hosch250** [comment]
I'm pretty sure some of those are reused in different places. Especially any with placeholders.

**Comment by retailcoder** [comment]
So what? It *is* the same message as far as UX / the user is concerned... no?

**Comment by Hosch250** [comment]
Yeah, coming from a different source. Source is more important than message.

**Comment by retailcoder** [comment]
Then prepend the source name to the resource key, and you have an ID 😄

**Comment by Hosch250** [comment]
Case in point: [Reorder params](https://github.com/rubberduck-vba/Rubberduck/blob/next/RetailCoder.VBE/Refactorings/ReorderParameters/ReorderParametersModel.cs#L75) and [Remove params](https://github.com/rubberduck-vba/Rubberduck/blob/next/RetailCoder.VBE/Refactorings/RemoveParameters/RemoveParametersModel.cs#L78).

Sure, that makes sense.

**Comment by retailcoder** [comment]
Wait a sec, no. Picture this (horrific) scenario:

If we have 200 inspections, and every time one completes, it pops a message box to report its execution time and how many results it issued: after the 5th one the user ticks "don't show this message again". And needs to do it 195 times. And another 5 next time inspections run.

A bit absurd of an example, but poses a real problem.

**Comment by Hosch250** [comment]
I think the caller needs to set the ID of the messagebox. Maybe something like "RemoveParams_NoTarget", "InpsectionExecutionTime", etc.

**Comment by bclothier** [comment]
Is this still relevant?

**Comment by retailcoder** [comment]
@bclothier yup!

**Comment by bclothier** [comment]
Should have thought of it sooner - in SSMS, I quite like how they have those extra buttons:
![image](https://user-images.githubusercontent.com/2367644/53215818-e2e02980-3617-11e9-9a3c-cefecdc4cd2b.png)

Though Windows messagebox are already copy-able, it makes for better discoverability and provide a way for linking documentation.
